pkg/storage/gcp/bucket_mock.go

```diff
@@ -0,0 +1,95 @@
+package gcp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+type bucketMock struct {
+	db   map[string][]byte
+	lock sync.RWMutex
+}
+
+func newBucketMock() Bucket {
+	b := &bucketMock{}
+	b.db = make(map[string][]byte)
+	return b
+}
+
+type bucketReader struct {
+	io.Reader
+	*bucketMock
+}
+
+type bucketWriter struct {
+	*bucketMock
+	path string
+}
+
+func (r *bucketReader) Close() error {
+	r.bucketMock.lock.RUnlock()
+	return nil
+}
+
+func (r *bucketWriter) Close() error {
+	r.bucketMock.lock.Unlock()
+	return nil
+}
+
+func (r *bucketWriter) Write(p []byte) (int, error) {
+	_, ok := r.bucketMock.db[r.path]
+	if !ok {
+		r.bucketMock.db[r.path] = make([]byte, 0)
+	}
+
+	r.bucketMock.db[r.path] = append(r.bucketMock.db[r.path], p...)
```

Comment:
Does this mean that if you call write on a file that exists it will append the contents?

Reply:
Yes. That was intended for handling writes that took more than one pass but has the side effect that different writers returned by different `Write` calls will append the content. 
Will change the `Write()` implementation to reset the map entry before returning a new `Writer`.

Reply:
:+1: Thanks, it would just keep more consistent as the GCP library. Using `Object.Write` overwrites any existing file with the same name.